### PR TITLE
Removed duplicate random devices

### PR DESCRIPTION
**Pull request by arkpar**
eth keeps ~130 handles to /dev/urandom opened. One for each cpp file that uses vector_ref.h

### CODE REVIEW

File: libdevcore/Common.cpp [hunk 1]
```diff
@@ -33,6 +33,9 @@ char const* Version = ETH_PROJECT_VERSION;
 
 const u256 UndefinedU256 = ~(u256)0;
 
+unsigned char s_cleanseCounter = 0;
+boost::random_device s_vectorRefEngine;
```
Comment (arkpar) [inline libdevcore/Common.cpp]: better place for these?

**Comment by gavofyork** [comment]
So `vector_ref` is meant to be header-only, hence the `static` rather than `extern`. We'll need to check that all usages of it link libdevcore and introduce `vector_ref.cpp` to merge this PR safely.

**Comment by arkpar** [comment]
vector_ref::randomize is not actually used anywhere except for SecureFixedHash::random. Refactored code for SecureFixedHash::random to use FixedHash::random and removed vector_ref::randomize

File: libdevcore/FixedHash.h [hunk 2]
```diff
@@ -151,16 +151,14 @@ class FixedHash
 
 	/// @returns a randomly-valued hash
 	template <class Engine>
-	static FixedHash random(Engine& _eng)
+	void randomize(Engine& _eng)
```
Comment (chfast) [inline libdevcore/FixedHash.h]: Doc comment is invalid.

**Comment by gavofyork** [comment]
no builds?

**Pull request merged**